conan/api/subapi/upload.py

```diff
@@ -40,3 +45,36 @@ def upload(self, package_list, remote):
         app.remote_manager.check_credentials(remote)
         executor = UploadExecutor(app)
         executor.upload(package_list, remote)
+
+    def upload_backup_sources(self, package_list):
+        app = ConanApp(self.conan_api.cache_folder)
+        config = app.cache.new_config
+        # TODO: Rethink this conf, does this live in core?
+        url = config.get("core.backup_sources:upload_url")
+        if url is None:
+            return
+        url = url if url.endswith("/") else url + "/"
+        download_cache_path = config.get("tools.files.download:download_cache")
+        download_cache_path = download_cache_path or config.get("core.download:download_cache")
+        if download_cache_path is None:
+            raise ConanException("Need to define 'core.download:download_cache'")
+
+        files = DownloadCache(download_cache_path).get_files_to_upload(package_list)
+        uploader = FileUploader(app.requester, verify=False, config=config)
+        # TODO: Dedup and list files
```

Comment:
```suggestion
        # TODO: List files
```

Reply:
Actually de-dup was finally not active. I was testing it and it could be problematic for other servers. It needs to be a new conf, but I think this optimization can wait a bit.